scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/NovaseqXProcessor.java

```diff
@@ -0,0 +1,241 @@
+package ca.on.oicr.gsi.runscanner.scanner.processor;
+
+import ca.on.oicr.gsi.runscanner.dto.IlluminaDragenNotificationDto;
+import ca.on.oicr.gsi.runscanner.dto.IlluminaNotificationDto;
+import ca.on.oicr.gsi.runscanner.dto.NotificationDto;
+import ca.on.oicr.gsi.runscanner.dto.type.AnalysisStatus;
+import ca.on.oicr.gsi.runscanner.dto.type.DRAGENWorkflow;
+import ca.on.oicr.gsi.runscanner.scanner.processor.NovaseqXProcessor.BCLConvertAnalysis.Analysis;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TimeZone;
+
+public class NovaseqXProcessor extends DefaultIllumina {
+  static class BCLConvertAnalysis {
+    List<Analysis> analyses = new LinkedList<>();
+
+    static class Analysis {
+      String sample;
+      int lane;
+      String read1File, read2File;
+      long readCount;
+
+      boolean isEmpty() {
+        return sample == null
+            && lane == 0
+            && read1File == null
+            && read2File == null
+            && readCount == 0;
+      }
+
+      ObjectNode toJson() {
+        ObjectNode analysisJson = MAPPER.createObjectNode();
+        analysisJson.put("Sample", sample);
+        analysisJson.put("Lane", lane);
+        analysisJson.put("Read1File", read1File);
+        analysisJson.put("Read2File", read2File);
+        analysisJson.put("ReadCount", readCount);
+        return analysisJson;
+      }
+    }
+
+    Analysis get(String sample, String lane) {
+      return get(sample, Integer.parseInt(lane));
+    }
+
+    Analysis get(String sample, int lane) {
+      return analyses
+          .stream()
+          .filter(a -> a.sample.equals(sample) && a.lane == lane)
+          .findFirst()
+          .orElse(new Analysis());
+    }
+
+    void put(Analysis newAnalysis) {
+      Analysis oldAnalysis = get(newAnalysis.sample, String.valueOf(newAnalysis.lane));
+      if (!oldAnalysis.isEmpty()) {
+        analyses.remove(oldAnalysis);
+      }
```

Comment:
The scan of the fastq_list file creates the first version of the Analysis object, then if there is a Demultiplex_Stats file, the updated Analysis objects created there need to overwrite the old ones

Reply:
- If there is a Demultiplex_Stats file, can we skip the fastq_list file? Or read Demultiplex_Stats first and skip parts of fastq_list that are unnecessary?
- What is the difference between these and reason for preferring Demultiplex_Stats?
- Do they get written at different times? Do we always expect both?

It would be good to document the logic here